lib/cocoapods/installer/xcode/pods_project_generator/pod_target_integrator.rb

```diff
@@ -211,14 +212,57 @@ def add_prepare_artifacts_script_phase(native_target, spec)
             end
           end
 
+          # Adds a script phase that copies and strips dSYMs that are part of this target. Note this only deals with
+          # vendored dSYMs.
+          #
+          # @param [PBXNativeTarget] native_target
+          #         the native target for which to add the the copy dSYM files build phase.
+          #
+          # @return [void]
+          #
+          def add_copy_strip_dsyms_script_phase(native_target)
+            dsym_paths = target.framework_paths.values.flatten.reject { |fmwk_path| fmwk_path.dsym_path.nil? }.map(&:dsym_path)
+
+            if dsym_paths.empty?
+              UserProjectIntegrator::TargetIntegrator.remove_copy_resources_script_phase_from_target(native_target)
+              script_phase = native_target.shell_script_build_phases.find { |bp| bp.name && bp.name.end_with?(UserProjectIntegrator::TargetIntegrator::COPY_DSYM_FILES_PHASE_NAME) }
+              native_target.build_phases.delete(script_phase) if script_phase.present?
+              return
+            end
+
+            phase_name = UserProjectIntegrator::TargetIntegrator::BUILD_PHASE_PREFIX + UserProjectIntegrator::TargetIntegrator::COPY_DSYM_FILES_PHASE_NAME
+            phase = UserProjectIntegrator::TargetIntegrator.create_or_update_shell_script_build_phase(native_target, phase_name)
+            phase.shell_script = dsym_script_contents(dsym_paths)
```

Comment:
this was done a bit differently than the other scripts in that it's not saved on disk, was that intentional? seems like we should be consistent

Reply:
The Pods manifest lock script phase is also directly added to it. I do not have a strong opinion though .......

Reply:
IMO less code embedded in the project itself the better, but I don't feel strongly enough about it to be against it in this case